### PR TITLE
acquire correct top-level lock at start of each step

### PR DESCRIPTION
**Pull request by swwilson-bsi**

### CODE REVIEW

File: src/ConnectorRunner.ts [hunk 1]
```diff
@@ -389,7 +391,7 @@ export class ConnectorRunner {
     const prevSynchConfigId = this._db.elements.queryElementIdByCode(LinkElement.createCode(this._db, IModel.repositoryModelId, "SynchConfig"));
     let idToReturn: string;
     if(prevSynchConfigId === undefined) {
-      await this._db.locks.acquireLocks({exclusive: IModel.dictionaryId});
+      await this._db.locks.acquireLocks({shared: IModel.rootSubjectId});
```
Comment (jackson-at-bentley) [inline src/ConnectorRunner.ts:394]: Why are we acquiring a shared lock to write the configuration link?

**Review by jackson-at-bentley — COMMENTED** [review]
I made similar changes in #55, but I'm locking `iModel.rootSubjectId` instead of `iModel.repositoryModelId` because I thought the new API operates on elements. It's a nonissue if `iModel.rootSubjectId === iModel.repositoryModelId`.

**Comment by swwilson-bsi** [comment]
Fine. If you already have these locking fixes in progress. Then we can delete my PR.

File: src/ConnectorRunner.ts (hunk 1 above)
Comment (jackson-at-bentley) [reply src/ConnectorRunner.ts:394]: Is a shared lock on a model equivalent to an exclusive lock on the modeled element?

**Pull request closed without merging**